src/main/java/io/kestra/storage/s3/S3Storage.java

```diff
@@ -332,11 +338,16 @@ public List<URI> deleteByPrefix(String tenantId, URI storagePrefix) throws IOExc
     }
 
     private String getPath(String tenantId, URI uri) {
+        if(uri == null) {
+            return "";
+        }
+
         parentTraversalGuard(uri);
+        String path = uri.getPath();
         if (tenantId == null) {
-            return uri.getPath();
+            return path;
         }
-        return "/" + tenantId + uri.getPath();
+        return "/" + tenantId + (path.startsWith("/") ? "" : "/") + path;
```

Comment:
path are always build by the `getPath` method, we may make sure that it starts with a / there to avoid checking it everywhere.